lib/helpers.bash

```diff
@@ -15,49 +15,49 @@ case "$OSTYPE" in
   'darwin'*) BASH_IT_SED_I_PARAMETERS=(-i "")
 esac
 
-function _command_exists ()
-{
-  _about 'checks for existence of a command'
-  _param '1: command to check'
-  _param '2: (optional) log message to include when command not found'
-  _example '$ _command_exists ls && echo exists'
-  _group 'lib'
-  local msg="${2:-Command '$1' does not exist!}"
-  if type -t "$1" &> /dev/null
-  then
-	  return 0
-  else
-	  _log_warning "$msg"
-	  return 1
-  fi
+function _command_exists() {
+	_about 'checks for existence of a command'
+	_param '1: command to check'
+	_param '2: (optional) log message to include when command not found'
+	_example '$ _command_exists ls && echo exists'
+	_group 'lib'
+	local msg="${2:-Command '$1' does not exist\!}"
+	if type "$1" &> /dev/null; then
+		return "$?"
```

Comment:
* Is there any value in passing types return code back to the caller?  Feels like it _should_ always be `0`.

More specifically:

1) If _our_ function's API is to return `0` or `1`, then we should probably just hard-code `0`, the same as we hard-code `1` in the error case.

2) If our function's API is to return 0 on success, non-0 on failure (ie error code with more context), then lets _also_ pass `$?` back in the failure case - ie. Document this function as a pass-through to `type -t` with additional logging, but returning type's error codes.

I tend to prefer no. 1 since our sister functions don't all use `type` so passing non-1 in the error cases may not be useful to the caller.

Reply:
Yes, this is inconsistent. I'll fix